great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -727,11 +727,7 @@ def update_datasource(
         Returns:
             The updated Datasource.
         """
-        if isinstance(datasource, FluentDatasource):
-            self._update_fluent_datasource(datasource=datasource)
-        else:
-            raise DataContextError("Datasource is not a FluentDatasource")  # noqa: TRY003
-        return datasource
```

Comment:
Returning the input datasource, rather that the output of `_update_fluent_datasource` resulted in a missing batch definition id.